src/tokens/saga.ts

```diff
@@ -376,12 +381,23 @@ export function* watchFetchBalance() {
 export function* watchAccountFundedOrLiquidated() {
   let prevTokenBalance
   while (true) {
-    const tokenBalance: ReturnType<typeof totalTokenBalanceSelector> = yield select(
+    const currentTokenBalance: ReturnType<typeof totalTokenBalanceSelector> = yield select(
       totalTokenBalanceSelector
     )
-    if (tokenBalance !== prevTokenBalance) {
-      // prevTokenBalance is undefined for the base case and null if token list
-      // is not yet loaded
+    const areTokenPricesStale: ReturnType<typeof stalePriceSelector> = yield select(
+      stalePriceSelector
+    )
+    const staleTokenBalance: ReturnType<typeof staleTokenBalanceSelector> = yield select(
+      staleTokenBalanceSelector
+    )
+    // we reset the usd value of all token balances to 0 if the exchange rate is
+    // stale, so it is okay to use stale token prices to monitor the account
+    // funded / liquidated status in this case
+    const tokenBalance = areTokenPricesStale ? staleTokenBalance : currentTokenBalance
```

Comment:
I think `staleTokenBalance` contains only the staled balance, instead of the total balance considering the staled price. So, it could happen that the staled balance is lower than `DOLLAR_MIN_AMOUNT_ACCOUNT_FUNDED`, triggering again the `account_liquidated` event, but the non-staled balance could be greater. Right?

Reply:
i'm not sure i follow - when the prices become stale, we override the existing usdPrice of the token to 0 so the token balance becomes 0. i added something to that selector, so that before we set the usdPrice to 0 i put the old usdPrice in the staleUsdPrice field.

so for sure the stale balance could be lower than the `DOLLAR_MIN_AMOUNT_ACCOUNT_FUNDED` but that means that the total balance before it became stale was also below this threshold so the event correctly fires? :/ am i missing something?

Reply:
fixed according to our discussion :) changed the stale price to be the last known price so that we don't have mixtures of stale and non stale prices